rfc/0018-multi-attribute-value-types/README.md

```diff
@@ -0,0 +1,162 @@
+<!--
+SPDX-FileCopyrightText: 2025 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0018: Multi attribute value types
+
+| | |
+|---|---|
+| Feature Tag | `multi-attribute value types` |
+| Status | `DISCUSSION` | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `tungstnballon` |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+This RFC allows value types to have multiple attributes.
+Value types are used to define tables, replacing the existing syntax.
+
+## Motivation
+
+- Mentioned in the 'Future Changes/Enhancements' section of RFC0014.
+- The jayvee interpreter still uses inheritance for value types, going against
+  the jayvee design principle of composition over inheritance.
+
+## Explanation
+
+Value types can contain multiple attributes:
+```jayvee
+valuetype Coordinate2D {
+  property x oftype decimal;
+  property y oftype decimal;
+}
+```
+
+
+### Usage
+
+Value types are used to define tables, each attribute defining a column:
+```jayvee
+valuetype Coordinate2D {
+  property x oftype decimal;
+  property y oftype decimal;
+}
+
+block toTable oftype TableInterpreter {
+  header: true;
+  schema: Coordinate2D;
+}
+```
+This replaces the current syntax:
+```jayvee
+block toTable oftype TableInterpreter {
+  header: true;
+  columns: [
+    "x" oftype decimal,
+    "y" oftype decimal,
+  ];
+}
+```
+
+Inline definition is also possible:
+```jayvee
+block toTable oftype TableInterpreter {
+  header: true;
+  schema: {
+    property x oftype decimal;
+    property y oftype decimal;
+  }
+}
+```
+
+One limitation with the new syntax is that column names have to match the jayvee
+identifier regex `/[_a-zA-Z][\w_]*/`, which excludes many expected names (e.g.
+with spaces).<br>
+The intended solution for this is a new composite block in the standard library,
+combining `CellWriter`, to rename the columns into a compatible format and
+`TableInterpreter`.<br>
+This follows the jayvee design principle of *libraries over language features*.
+
+Conceptually, a value type defines a table. The instance of that value type is a
+row inside the table.
+
+Nested value types are allowed:
+```jayvee
+valuetype Coordinate2D {
+  property x oftype decimal;
+  property y oftype decimal;
+}
+valuetype Circle {
+  property center oftype Coordinate2D;
+  property radius oftype decimal;
+}
+```
+
+At first, the jayvee interpreter will only implement nested value types, where the
```

Comment:
Are we using some `.value` magic naming? Or is the reference still by the name of the property? What does "implement" mean, e.g. what is the intended behavior otherwise? If it is "it crashes", this should be part of the RFC.